src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp

```diff
@@ -222,6 +222,8 @@ class TPackageScanner final : public AbstractHandler<std::shared_ptr<TScanContex
             }
         }
 
+        cnaNameBase = cnaName;
```

Comment:
Improve: This assignation won't have effect if the method returns at line 219.

Another option is to return a pair instead of having a return and an out parameter at the same time.

Reply:
I just noticed that it was the cause of the key = ' ' mentioned in the call. I didn't want to change the requirement of using a tuple but I can change that

Reply:
I found complex to make a decision about what value to use after we return the pair.

Reply:
> I just noticed that it was the cause of the key = ' ' mentioned in the call.

Good catch then, I didn't see it in the morning.

> I didn't want to change the requirement of using a tuple but I can change that
I found complex to make a decision about what value to use after we return the pair.

It is just an option. The important thing is to always return a value.

I'll leave here the example I used locally for the tests.
It is a reference of what I meant, there are many ways to solve this. I thought it might be confusing to return values with two different ways at the same time

<details><summary>Details</summary>
<p>

```c++
    std::pair<std::string, std::string> getCNA(std::shared_ptr<TScanContext> ctx)
    {
        auto cnaName {m_databaseFeedManager->getCnaNameByFormat(ctx->packageFormat().data())};

        if (cnaName.empty())
        {
            cnaName = m_databaseFeedManager->getCnaNameBySource(ctx->packageSource().data());
            if (cnaName.empty())
            {
                cnaName =
                    m_databaseFeedManager->getCnaNameByPrefix(ctx->packageVendor().data(), ctx->osPlatform().data());
                if (cnaName.empty())
                {
                    cnaName = m_databaseFeedManager->getCnaNameByContains(ctx->packageVendor().data(),
                                                                          ctx->osPlatform().data());
                    if (cnaName.empty())
                    {
                        return {DEFAULT_CNA, DEFAULT_CNA};
                    }
                }
            }
        }
        const auto mapping = TGlobalData::instance().cnaMappings();

        const auto& cnaMapping = mapping.at("cnaMapping");
        const auto platformEquivalence = [&](const std::string& platform) -> const std::string&
        {
            const auto& platformMapping = mapping.at("platformEquivalence");
            if (const auto it = platformMapping.find(platform); it == platformMapping.end())
            {
                return platform;
            }
            else
            {
                return it->template get_ref<const std::string&>();
            }
        };

        const auto majorVersionEquivalence = [&](const std::string& platform,
                                                 const std::string& majorVersion) -> const std::string&
        {
            const auto& majorVersionMapping = mapping.at("majorVersionEquivalence");
            if (const auto itPlatform = majorVersionMapping.find(platform); itPlatform == majorVersionMapping.end())
            {
                return majorVersion;
            }
            else
            {
                if (const auto itMajorVersion = itPlatform->find(majorVersion); itMajorVersion == itPlatform->end())
                {
                    return majorVersion;
                }
                else
                {
                    return itMajorVersion->template get_ref<const std::string&>();
                }
            }
        };

        if (const auto it = cnaMapping.find(cnaName); it == cnaMapping.end())
        {
            return {cnaName, cnaName};
        }
        else
        {
            std::string base = it->template get<std::string>();
            Utils::replaceAll(base, "$(PLATFORM)", platformEquivalence(ctx->osPlatform().data()));
            Utils::replaceAll(base,
                              "$(MAJOR_VERSION)",
                              majorVersionEquivalence(ctx->osPlatform().data(), ctx->osMajorVersion().data()));
            return {cnaName, base};
        }
    }
```

</p>
</details>

Reply:
Changed

Reply:
Agree with the use of a pair, a tuple was proposed just because was the first thing that poped into our minds